packaging/_tokenizer.py

```diff
@@ -0,0 +1,154 @@
+import re
+from typing import Dict, Generator, NoReturn, Optional
+
+from .specifiers import Specifier
+
+
+class Token:
+    def __init__(self, name: str, text: str, position: int) -> None:
+        self.name = name
+        self.text = text
+        self.position = position
+
+    def matches(self, name: str = "", text: str = "") -> bool:
```

Comment:
The `text` argument is now unused, can you remove it?

 If yes, `match_args`/`match_kwargs` below can be replaced by `name`.